src/main/java/de/tum/in/www1/artemis/domain/Feedback.java

```diff
@@ -452,6 +453,15 @@ public double computeTotalScore(double inputScore, Map<Long, Integer> gradingIns
         return totalScore;
     }
 
+    /**
+     * Converts this entity into a {{@link FeedbackDTO}} object
+     *
+     * @return the converted DTO
+     */
+    public FeedbackDTO toFeedbackDTO() {
+        return new FeedbackDTO(getText(), getDetailText(), getHasLongFeedbackText(), getReference(), getCredits(), isPositive(), getType(), getVisibility());
+    }
+
```

Comment:
I choose this approach since we can make use of inheritance, (see toSubmissionDTO() in Submission and ProgrammingSubmission).

Reply:
(opinion) I prefer to have an additional constructor (or `of(domainObject)`-method) in the DTO that takes the domain object as parameter rather than constructing the DTO from within the domain object.

Reasoning:

- The domain object should only be concerned about the representation of its own data. 
  - The DTO(s) that contain only different parts of that data should be responsible to extract the relevant parts from the full object since they know best which of the data is actually relevant.
  - In case of multiple DTOs for the same domain object that extract different parts of the data (e.g. `ResultForStudent` and `ResultForInstructor` where for one secret information is removed, in the other one not) each DTO remains responsible to fill the data in correctly only for its own representation, rather than the domain object having to have knowledge which elements should be filled in which way for all of the related DTOs.
- Architecture: domain classes should not depend on representation classes in the higher-level `web.rest` package.